app/models/course/assessment/question/programming.rb

```diff
@@ -110,22 +110,40 @@ def non_autograded_template_files=(template_files)
 
   private
 
+  # Create new package or re-evaluate the old package.
+  def process_package
+    return if @duplicating
```

Comment:
Shall we have `attr_reader :duplicating` and then do a `before_save :process_package, unless: :duplicating?`